### PR TITLE
feat(build): community_scripts toggle in mission.yaml

### PR DESCRIPTION
**Pull request by davidp57**
## Summary
- `get_community_script_files()` refactored to return `list[dict]` with stable `id`/`path`/`dest` keys (COMM-001)
- `MissionBuilderWorker` parses a new `community_scripts:` section in `mission.yaml`; absent section keeps all scripts active (COMM-002/003)
- Filter applied to both static (`insert_veaf_trigrules`) and dynamic (`insert_veaf_triggers`) trigger generation
- Default `mission.yaml` and both YAML reference docs updated with the new section (COMM-004/005)
- 10 new unit tests covering parsing, filtering, empty set, and unknown-id handling (COMM-006)

## Test plan
- [ ] `poetry run pytest test/python/mission_builder/ --no-cov` → 269 passed
- [ ] Build a mission with `community_scripts: { skynet: { enabled: false } }` — verify `skynet-iads-compiled.lua` absent from the `.miz`
- [ ] Build without `community_scripts:` section — verify all scripts present as before

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Add configurable toggling of community Lua scripts in missions via a new mission.yaml section and wire it through script collection and trigger generation.

New Features:
- Introduce a community_scripts section in mission.yaml to enable or disable individual community Lua scripts by stable identifier.
- Support per-mission filtering of community scripts in both static and dynamic trigger generation based on configured enablement flags.

Enhancements:
- Refactor community script descriptors to use stable id/path/dest dictionaries instead of tuples and centralize filtering through a helper.
- Warn on invalid community_scripts configuration values and unknown script identifiers while keeping defaults unchanged when the section is absent.

Documentation:
- Document the community_scripts section and available script identifiers in both English and French mission YAML reference docs, and add commented examples to the default mission.yaml.

Tests:
- Add unit tests covering community_scripts parsing, enabling/disabling behavior, handling of empty and invalid configurations, and active script selection helper behavior.

Chores:
- Update backlog and task tracking statuses to mark related COMM, CUSTOM, RADIO, and REV items as completed.
- Record the new community_scripts configuration capability in the changelog.

### CODE REVIEW

File: src/python/veaf-tools/mission_builder/mission_builder_worker.py [hunk 1]
```diff
@@ -190,6 +190,31 @@ def __init__(
...
+        comm_raw = self.mission_yaml.get("community_scripts")
+        if comm_raw is not None and not isinstance(comm_raw, dict):
+            logger.warning(
+                f"'community_scripts' in mission.yaml must be a mapping (got {type(comm_raw).__name__}); ignoring."
+            )
+            comm_raw = None
+        if comm_raw is None:
+            self.enabled_community_script_ids: set[str] | None = None
+        else:
+            all_ids = {s["id"] for s in get_community_script_files()}
```
[the remark this answers is not in the document]
Comment (davidp57) [reply src/python/veaf-tools/mission_builder/mission_builder_worker.py]: Fixed: empty {} now treated as absent (all active); parsing switched to opt-out semantics — only explicitly disabled entries are removed from the full set.

File: test/python/mission_builder/test_community_scripts_toggle.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        worker = _make_worker(yaml)
+        self.assertEqual(worker.enabled_community_script_ids, ALL_IDS)
+
+    def test_one_script_disabled(self) -> None:
+        """A single script with enabled: false is absent from enabled_community_script_ids."""
+        worker = _make_worker("community_scripts:\n  ctld: {enabled: false}\n")
+        assert worker.enabled_community_script_ids is not None
+        self.assertNotIn("ctld", worker.enabled_community_script_ids)
+
+    def test_multiple_scripts_disabled(self) -> None:
```
[the remark this answers is not in the document]
Comment (davidp57) [reply test/python/mission_builder/test_community_scripts_toggle.py:55]: Added tests for boolean shorthand (true/false), null value, empty dict per-script entry, and opt-out semantics.

File: test/python/mission_builder/test_community_scripts_toggle.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+class TestActiveCommunityScripts(unittest.TestCase):
+    """Unit tests for _active_community_scripts helper."""
+
+    def test_none_returns_all(self) -> None:
+        """_active_community_scripts returns all scripts when enabled_community_script_ids is None."""
+        worker: MissionBuilderWorker = object.__new__(MissionBuilderWorker)
+        worker.enabled_community_script_ids = None
+        result = worker._active_community_scripts()
+        self.assertEqual(result, get_community_script_files())
+
```
[the remark this answers is not in the document]
Comment (davidp57) [reply test/python/mission_builder/test_community_scripts_toggle.py:118]: Added test_disabled_script_absent_from_active_scripts_and_paths which verifies the YAML→enabled_set→_active_community_scripts flow end-to-end; get_collected_community_script_files requires real files on disk so not covered at this level.

File: doc/MISSION_YAML_REFERENCE.md [hunk 4]
```diff
@@ -299,6 +299,39 @@ Voir les pages respectives pour le schéma complet :
 
 ---
 
+### `community_scripts:`
+
+Permet d'activer ou désactiver individuellement les scripts Lua communautaires (MIST, CTLD, CSAR, etc.) injectés dans la mission. Lorsque cette section est absente, **tous** les scripts communautaires sont inclus.
```
[the remark this answers is not in the document]
Comment (davidp57) [reply doc/MISSION_YAML_REFERENCE.md]: Fixed: 'ou de désactiver'.
